pkg/kubectl/metricsutil/metrics_printer.go

```diff
@@ -53,16 +53,20 @@ func NewTopCmdPrinter(out io.Writer) *TopCmdPrinter {
 	return &TopCmdPrinter{out: out}
 }
 
-func (printer *TopCmdPrinter) PrintNodeMetrics(metrics []metricsapi.NodeMetrics, availableResources map[string]v1.ResourceList, noHeaders bool) error {
+func (printer *TopCmdPrinter) PrintNodeMetrics(metrics []metricsapi.NodeMetrics, availableResources map[string]v1.ResourceList, noHeaders bool, sortBy string) error {
 	if len(metrics) == 0 {
 		return nil
 	}
 	w := printers.GetNewTabWriter(printer.out)
 	defer w.Flush()
 
-	sort.Slice(metrics, func(i, j int) bool {
-		return metrics[i].Name < metrics[j].Name
-	})
+	if len(sortBy) > 0 {
+		sortNodesByMetrics(metrics, sortBy)
+	} else {
+		sort.Slice(metrics, func(i, j int) bool {
+			return metrics[i].Name < metrics[j].Name
+		})
+	}
```

Comment:
Why not just implementing `sort.Interface` (https://golang.org/pkg/sort/#Interface) and pass parameters, in it you'd act accordingly whether or not `sortBy` was passed. Have a look at https://github.com/kubernetes/kubernetes/blob/a93f803f8e400f1d42dc812bc51932ff3b31798a/pkg/kubectl/cmd/get/sorter.go#L364